src/Xamarin.Android.Build.Tasks/Tasks/CheckForInvalidResourceFileNames.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using Xamarin.Android.Tasks;
+
+namespace Xamarin.Android.Tasks {
+	public class CheckForInvalidResourceFileNames : Task {
```

Comment:
I thought we'd discussed/considered replacing all invalid Android resource filename characters with `_`...

Reply:
I know we did. I did it this way to maintain backward compatibility with `aapt`.